graphql-cli/src/main/java/io/ballerina/graphql/cmd/GraphqlCmd.java

```diff
@@ -168,57 +197,119 @@ private void validateInputFlags() throws CmdException {
+                filePath.endsWith(BAL_EXTENSION) || filePath.endsWith(GRAPHQL_EXTENSION);
+    }
+
+    private boolean isModeCompatible() throws CmdException {
+        String filePath = argList.get(0);
+        if ((mode == null || MODE_CLIENT.equals(mode))) {
+            return filePath.endsWith(Constants.YAML_EXTENSION) || filePath.endsWith(Constants.YML_EXTENSION);
+        } else if (MODE_SCHEMA.equals(mode)) {
+            return filePath.endsWith(Constants.BAL_EXTENSION);
+        } else if (MODE_SERVICE.equals(mode)) {
+            return filePath.endsWith(Constants.GRAPHQL_EXTENSION);
+        } else {
+            throw new CmdException(String.format(MESSAGE_FOR_INVALID_MODE, mode));
+        }
     }
 
     /**
      * Execute the correct operation according to the given inputs.
      *
-     * @throws CmdException               when a graphql command related error occurs
-     * @throws ParseException             when a parsing related error occurs
-     * @throws IOException                If an I/O error occurs
-     * @throws GenerationException        when a graphql client generation related error occurs
-     * @throws ValidationException        when validation related error occurs
-     * @throws SchemaFileGenerationException  when a SDL schema generation related error occurs
+     * @throws CmdException                  when a graphql command related error occurs
+     * @throws ParseException                when a parsing related error occurs
+     *                                       if (filePath.endsWith(YAML_EXTENSION) || filePath.endsWith(YML_EXTENSION))
+     *                                       {
+     *                                       generateClient(filePath);
+     *                                       } else if (filePath.endsWith(BAL_EXTENSION)) {
+     *                                       generateSchema(filePath);
+     *                                       } else if (filePath.endsWith(GRAPHQL_EXTENSION)) {
+     *                                       generateService(filePath);
+     *                                       } else {
+     *                                       throw new CmdException(MESSAGE_FOR_INVALID_FILE_EXTENSION);
+     *                                       }@throws IOException                   If an I/O error occurs
+     * @throws GenerationException           when a graphql client generation related error occurs
+     * @throws ValidationException           when validation related error occurs
+     * @throws SchemaFileGenerationException when a SDL schema generation related error occurs
      */
     private void executeOperation()
             throws CmdException, ParseException, IOException, ValidationException, GenerationException,
             SchemaFileGenerationException {
         String filePath = argList.get(0);
-        if (filePath.endsWith(YAML_EXTENSION) || filePath.endsWith(YML_EXTENSION)) {
+
+        if (mode == null || MODE_CLIENT.equals(mode)) {
```

Comment:
Is this logic correct? What happens if the `mode == null` & input file is a `.bal` file or a '.graphql' file?

Reply:
Since `mode` will have new values in the future. It was decided to consider `mode` as a compulsory flag except for client generation. Considering client generation is the default behaviour if `mode` flag is omitted. 
So for the above scenarios, an exception will be thrown indicating `mode` value `null` is not allowed to use with `.bal` or `.graphql` file.

Reply:
In that case, we have to update all the documentation related to the GraphQL tool since the `mode` flag changes the SDL schema generation command as well.

Reply:
suggestion:
This is to take an idea. Isn't it better to make the `mode` flag compulsory only when two different operations take the same type of input?
Since we have 3 different input file types for client generation, schema generation, and service generation, We can make the mode flag optional. In that case the, for the command `bal graphql -i <input_file>`
1. if the input file is `.yml` file, generate the `client`.
2. if the input file is `.bal` file, generate the `schema`.
3. if the input file is `.graphql`, generate the `service`.

Here we can consider these three operations as the default behavior of the tool. If we have a new feature that takes one of the above input file types, we can make the `mode` flag compulsory for that.

With this suggestion, following can be achieved.
1. Can make these three operations the default behavior of the tool.
2. Make the command short.
3. Reduce the number of options that can take as the `mode`.

@ThisaruGuruge WDYT?